paasta_tools/vitesscluster_tools.py

```diff
@@ -72,21 +662,98 @@ def validate(
             return []
 
 
-def load_vitess_instance_config(
+def generate_vitess_instance_config(
+    instance_config: Dict,
+) -> Dict:
+    # Generate the vitess instance config from the yelpsoa config
+
+    cpus = instance_config.get("cpus")
+    mem = instance_config.get("mem")
+    deploy_group = instance_config.get("deploy_group")
+    zk_address = instance_config.get("zk_address")
+    paasta_pool = instance_config.get("paasta_pool")
+    paasta_cluster = instance_config.get("paasta_cluster")
+    cells = instance_config.get("cells")
+    keyspaces = instance_config.get("keyspaces")
+    region = instance_config.get("region")
+
+    vitess_instance_config = {
+        "namespace": "paasta-vitessclusters",
+        "cpus": cpus,
+        "mem": mem,
+        "min_instances": 1,
+        "max_instances": 1,
+        "deploy_group": deploy_group,
+        "autoscaling": {"setpoint": 0.7},
+        "env": {
+            "OPERATOR_NAME": "vitess-operator",
+            "POD_NAME": "vitess-k8s",
+            "PS_OPERATOR_POD_NAME": "vitess-k8s",
+            "PS_OPERATOR_POD_NAMESPACE": "paasta-vitessclusters",
+            "WATCH_NAMESPACE": "paasta-vitessclusters",
+        },
+        "healthcheck_grace_period_seconds": 60,
+        "healthcheck_mode": "cmd",
+        "healthcheck_cmd": "true",
+        "images": {
+            "vtctld": VTCTLD_IMAGE,
+            "vtadmin": VT_ADMIN_IMAGE,
+            "vtgate": VT_GATE_IMAGE,
+            "vttablet": VT_TABLET_IMAGE,
+        },
+        "globalLockserver": {
+            "external": {
+                "implementation": TOPO_IMPLEMENTATION,
+                "address": zk_address,
+                "rootPath": TOPO_GLOBAL_ROOT,
+            }
+        },
+        "cells": [
+            build_cell_config(cell, paasta_pool, paasta_cluster, region)
+            for cell in cells
+        ],
+        "vitessDashboard": build_vitess_dashboard_config(
+            cells, paasta_pool, paasta_cluster, zk_address
+        ),
+        "vtadmin": build_vt_admin_config(cells, paasta_pool, paasta_cluster),
+        "keyspaces": build_keyspaces_config(
+            cells, keyspaces, paasta_pool, paasta_cluster, zk_address, region
+        ),
+        "updateStrategy": {"type": "Immediate"},
```

Comment:
hmm, is this a statefulset update strategy? https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#update-strategies only documents OnDelete and RollingUpdate

Reply:
It's not a statefulset update strategy but specific to VitessKeyspaces CRD understood by the operator which only has two options https://github.yelpcorp.com/misc/compute-infra-k8s/blob/master/clusters/eks-infrastage/90-vitess-operator/eks/40-crd-planetscale.com-vitesskeyspaces.yaml#L1074

Reply:
ah, gotcha :)